### ISSUE TITLE
Showing which subscriptions are working on a particular page

### ISSUE THREAD

**Issue by privacymaidenfleeinginfovampires** [RequestPolicyContinued/requestpolicy#660]
I'd like to be able to see which subscriptions are providing rules for the page I'm viewing. I tend to lean on the side of broken web over surveillance web, and I would like to make sure any particular subscription isn't being too permissive.

I know that there are issues discussing displaying the whole of a subscription, but I think having the context of a webpage makes them a bit easier to read.

**Comment by nodiscc** [comment]
~~You can see a proposition to see subscriptions rules directly in the menu at https://github.com/RequestPolicyContinued/requestpolicy/issues/601~~ _no, forgot to add it to the menu mockup_. It was suggested in #98 to have a marker on menu items if they are affected by subscriptions.

Eg if a subscription allows requests to `gimmepersonaldata.net`, the [allowed icon](https://github.com/RequestPolicyContinued/requestpolicy/blob/dev-1.0/src/skin/menu-allowed.svg) for this domain in the menu would have an overlayed green dot added.

Edit: found in https://github.com/RequestPolicyContinued/requestpolicy/issues/509

![](https://cloud.githubusercontent.com/assets/3950390/4937080/4d1fcdfa-65c0-11e4-8964-34ee1e2b3459.png)

@privacymaidenfleeinginfovampires would this work for you? Can we continue discussion in #509 ?

**Comment by privacymaidenfleeinginfovampires** [comment]
#509 seems more about differentiating user policy from subscription policy. What I want is a way to differentiate subscription policies from each other, in the hope that there will eventually be more than one source for subscriptions.

Admittedly, it's easy to imagine the interface becoming cluttered. I don't know how feasible this idea actually is.

**Comment by nodiscc** [comment]
> #509 seems more about differentiating user policy from subscription policy.

Yes.

> What I want is a way to differentiate subscription policies from each other, in the hope that there will eventually be more than one source for subscriptions.

This is interesting. How/where could this info be displayed? As you said, we should be very careful about interface clutter. Maintaining subscriptions is also a very tedious task,even with a [single, centralized list](https://github.com/RequestPolicyContinued/subscriptions).

**Comment by jdgalt** [comment]
What I'd really like is an analysis mode that shows me the unfiltered web page and lets me identify just what sites need to be unblocked to allow me to see or use a particular feature on that page.

washingtonpost.com is my best example of a site that needs this treatment.  It's a real Charlie Foxtrot that references at least 20 domains, of which I only need 5 (but spent a whole afternoon figuring out which ones they were).

**Comment by ansell** [comment]
It isn't technically possible to know which of the included scripts are relevant to each element on a page as any script on a page can change any element on the page, and some scripts may depend on other scripts being loaded first to function properly.

However, it could be possible to show which domains images and other interactive elements come from and unblock them using a right click on the blocked image elements. NoScript and AdBlock Plus already do that for blocked images/applets/flash/etc.

**Comment by nodiscc** [comment]
> show which domains images and other interactive elements come from and unblock them using a right click on the blocked image elements

https://github.com/RequestPolicyContinued/requestpolicy/issues/65 and https://github.com/RequestPolicyContinued/requestpolicy/issues/129

**Comment by myrdd** [comment]
> What I want is a way to differentiate subscription policies from each other (…).
> (…)
> I don't know how feasible this idea actually is.

Yes, currently there's only one subscription source, the official one. Yes, I already had the idea in my mind – the idea of making subscription sources configurable.

Besides that, I like your idea. How about a *letter* which overlays the flag. Imagine this scenario:

- You install RPC and get official subscriptions with it. That official subs is assigned the letter *„o“* (short for *official*).
- You add another subscription source, and you assign it the letter *„a“*.
- When you visit a website, and a subscription rule applies to a request, then the letter will be layed over the green *allowed* icon (![green-allowed-icon](https://cloud.githubusercontent.com/assets/4286829/7812222/a03aeb08-03b0-11e5-85d2-76fb9acf76d1.png)).

In other words: if you only have the official subscriptions, then the ovelay will be like in @nodiscc's mockup (https://github.com/RequestPolicyContinued/requestpolicy/issues/660#issuecomment-105240923). As soon as you add more subscriptions, the letters will be used instead of the "dot". What do you think @privacymaidenfleeinginfovampires?

-------------------------

> What I'd really like is an analysis mode that shows me the unfiltered web page and lets me right-click on an element to identify just what sites need to be unblocked to allow me to see or use that element.

@jdgalt I like your suggestion. Could you take a look at #536 and add your comment there?

------------------------

> #65 and #129

@nodiscc isn't one of them a duplicate?

**Comment by privacymaidenfleeinginfovampires** [comment]
@myrdd Honestly, it sounds like it would have a lot of unpleasant edge-cases. I think the first-click-menu doesn't need to do more than differentiate between user rules and subscription rules. Then, if you click on one of the items in the list with the 'dot', the subscription name could appear below the 'create temporary rule' list.

**Comment by myrdd** [comment]
> Then, if you click on one of the items in the list with the 'dot', the subscription name could appear below the 'create temporary rule' list.

Sounds like a good option. So we should rename the issue, e.g. to *„in the menu, when a subscription rule matches, display the name of the respective subscription list“*.
